### PR TITLE
Updated DG documentation

### PR DESCRIPTION
**Pull request by Tan-Jia-Rong**
Edited search and navigation manual testing for consistency
Updated wrong usage of `list` to `list-module`

The rest remains untouched. However, due to merge conflict, the files changes might seen bit confusing at the bottom few codes.

I recommend that we do an full read through of our user guide and documentation guide on our website today as a team and fix all other possible errors we may see before doing 1 last PR (if needed) and submit our documents

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1417,103 +1418,94 @@ Remember to perform a backup before deletion if necessary. Then, open Plannit.
...
+        * Module list is updated with module code starting with `CS2103T`(case-insensitive).
         * Number of modules listed shown in the status message.
 
-3. Other incorrect `find-module` usage to try:
-    * Finding modules while not at home page <br>
-      Expected:
-        * Module list remains the same.
-        * Command box remains the same.
-        * Error details shown in the status message.
+3. Finding modules while not at home page
```
Comment (teoyuqi) [inline docs/DeveloperGuide.md:1452]: Perhaps you could add "not at home page" as a prerequisite?

**Review by teoyuqi — COMMENTED** [review]
Just a minor nitpick

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Tan-Jia-Rong) [reply docs/DeveloperGuide.md:1452]: I did not add it in as it was in the sentence itself. However it is a good suggest. I will add it in for the rest as well thanks for raising this up!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1723,72 +1714,65 @@ Prerequisites:
...
-        * Must be at the home page.
+   1. Prerequisites: Must be at the home page. Multiple persons in Plannit.
 
-    1. Test case: `find-person Alex`<br>
-       Expected:
-        * Person list is updated with persons whose names starting with `Alex`<br>(case-insensitive).
-        * Number of persons listed shown in the status message.
+   2. Test case: `find-person Alex`<br>
+      Expected:
+      * Person list is updated with persons whose names starting with `Alex`(case-insensitive).
```
Comment (cheeheng) [inline docs/DeveloperGuide.md:1723]:
```suggestion
      * Person list is updated with persons whose names start with `Alex`(case-insensitive). 
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -1723,72 +1714,65 @@ Prerequisites:
...
-3. Other incorrect `find-person` usage to try:
-    * Finding persons while not at home page <br>
-      Expected:
-        * Person list remains the same.
-        * Command box remains the same.
-        * Error details shown in the status message.
+    2. Test case: `find-person Bernice`<br>
+        Assumption: Person `Bernice` is not in the current person list<br>
+        Expected:
+         * Person list is updated with persons whose names starts with `Bernice` (case-insensitive).
```
Comment (cheeheng) [inline docs/DeveloperGuide.md:1745]:
```suggestion
         * Person list is updated with persons whose names start with `Bernice` (case-insensitive).
```

**Review by cheeheng — COMMENTED** [review]
LGTM except for grammatical error nitpicks.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Tan-Jia-Rong) [reply docs/DeveloperGuide.md:1729]: thanks for catching all my grammatical errors as always appreciate it!

**Review by cheeheng — APPROVED** [review]
LGTM

**Review by teoyuqi — APPROVED** [review]
LGTM

**Pull request merged**